include/bout/mesh.hxx

```diff
@@ -444,11 +444,26 @@ class Mesh {
       // No coordinate system set. Create default
       // Note that this can't be allocated here due to incomplete type
       // (circular dependency between Mesh and Coordinates)
-      coords_map.emplace(location, createDefaultCoordinates(location));
+      coords_map[location] = std::shared_ptr<Coordinates>(nullptr);
+      coords_map[location] = createDefaultCoordinates(location);
       return coords_map[location].get();
     }
   }
 
+  /// Re-calculate staggered Coordinates, useful if CELL_CENTRE Coordinates are changed
+  void recalculateStaggeredCoordinates() {
```

Comment:
This seems to be separate to the original goal of this PR (to prevent hangs) -- it's obviously important that we think about how to ensure our coordinates are consistent but this could involve a number of changes and requires more design decisions which will likely hold up the original bug fix. This is probably fine as we have an alternative fix for master, but just thought I'd mention it.

Reply:
:+1: I'd agree it's better to move this discussion to another PR/Issue, to avoid delaying and complicating this one.